File: arosaje-backend/src/main/java/fr/epsi/mspr/arosaje/entity/Photo.java
package fr.epsi.mspr.arosaje.entity;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "photos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Photo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @Lob
    @Column(name = "image_blob", columnDefinition="LONGBLOB")
    private byte[] imageBlob;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "commentaire_id")
    private TicketComment ticketComment;

    @ManyToOne
    @JoinColumn(name = "plant_id", referencedColumnName = "id")
    private Plant plant;
}

File: arosaje-frontend/src/Domain/Photo.js
class Photo{
    id: number;
    user: User;
    referenceId: number;
    referenceType: string;
    imageBlob: Blob;
    createdAt: Date;

    constructor(id: number, user: User, referenceId: number, referenceType: string, imageBlob: Blob, createdAt: Date) {
        this._id = id;
        this._user = user;
        this._referenceId = referenceId;
        this._referenceType = referenceType;
        this._imageBlob = imageBlob;
        this._createdAt = createdAt;
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get user(): User {
        return this._user;
    }

    set user(value: User) {
        this._user = value;
    }

    get referenceId(): number {
        return this._referenceId;
    }

    set referenceId(value: number) {
        this._referenceId = value;
    }

    get referenceType(): string {
        return this._referenceType;
    }

    set referenceType(value: string) {
        this._referenceType = value;
    }

    get imageBlob(): Blob {
        return this._imageBlob;
    }

    set imageBlob(value: Blob) {
        this._imageBlob = value;
    }

    get createdAt(): Date {
        return this._createdAt;
    }

    set createdAt(value: Date) {
        this._createdAt = value;
    }
}